### ISSUE TITLE
Issue with go-ipfs-cmdkit

### ISSUE THREAD

**Issue by hsanjuan** [ipfs-inactive/dev-team-enablement#132]
## Which Project is this about?

Standard Go project

## Provide a link to a example run where it went wrong

https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fgo-ipfs-cmdkit/detail/master/3/pipeline/

## Which platform did the issue happen on?

All.

Cannot check out the repo correctly:

`package github.com/ipfs/go-ipfs-cmdkit/files: /tmp/jenkins/workspace/IPFS_go-ipfs-cmdkit_master-M37TKJJVWCGNVQ7NTTGYMWY2WGHUSBCQG6VQBBGMYYOULF6DLTVQ/src/github.com/ipfs/go-ipfs-cmdkit exists but /tmp/jenkins/workspace/IPFS_go-ipfs-cmdkit_master-M37TKJJVWCGNVQ7NTTGYMWY2WGHUSBCQG6VQBBGMYYOULF6DLTVQ/src/github.com/ipfs/go-ipfs-cmdkit/.git does not - stale checkout?`

**Comment by Stebalien** [comment]
@VictorBjelkholm this is still going on. Is something not being cleaned up correctly?

**Comment by VictorBjelkholm** [comment]
@Stebalien I don't think it has anything to do with cleaning up properly as each build is in a fresh directory.

I do think it has something to do with the GOPATH setup we do in the beginning of Golang jobs (https://github.com/ipfs/jenkins-libs/blob/master/vars/golang.groovy#L28-L33) but unsure exactly where the issue is as it seems to just affect go-ipfs-cmdkit and no other jobs.

**Comment by magik6k** [comment]
It looks like it's caused by gx depending on go-ipfs-cmdkit and us invoking `go get [...]/gx`:

```
+ go get -v github.com/whyrusleeping/gx

github.com/whyrusleeping/gx (download)

github.com/blang/semver (download)

github.com/mitchellh/go-homedir (download)

github.com/urfave/cli (download)

github.com/ipfs/go-ipfs-api (download)

github.com/ipfs/go-ipfs-cmdkit (download)

package github.com/ipfs/go-ipfs-cmdkit/files: /tmp/jenkins/workspace/IPFS_go-ipfs-cmdkit_master-M37TKJJVWCGNVQ7NTTGYMWY2WGHUSBCQG6VQBBGMYYOULF6DLTVQ/src/github.com/ipfs/go-ipfs-cmdkit
exists but /tmp/jenkins/workspace/IPFS_go-ipfs-cmdkit_master-M37TKJJVWCGNVQ7NTTGYMWY2WGHUSBCQG6VQBBGMYYOULF6DLTVQ/src/github.com/ipfs/go-ipfs-cmdkit/.git
does not - stale checkout?

```

**Comment by Stebalien** [comment]
Gah. We need to just fetch gx from dists.

**Comment by hsanjuan** [comment]
Related https://github.com/ipfs/testing/issues/137

**Comment by daviddias** [comment]
Isn't #137 the exact same issue? Shall we close one of them?
